packages/app/src/specs/generators/empty/EmptyGeneratorCardStepOne.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <span>
+    <div class="p-24px">
+      <Input
+        v-model="specFile"
+        placeholder="Enter spec file ..."
+      >
+        <template #prefix>
+          <i-cy-document-blank_x16 class="icon-light-gray-50 icon-dark-gray-300 min-w-16px min-h-16px" />
+        </template>
+      </Input>
+
+      <div
+        v-if="!isValidSpecFile"
+        class="mt-16px bg-red-200"
+      >
+        Your filename is invalid because it doesn't match the following <b>specPattern</b>.
+      </div>
+
+      <div class="mt-16px">
+        <SpecPatterns
+          :gql="{...props.gql}"
+        />
+      </div>
+    </div>
+
+    <StandardModalFooter class="flex gap-16px">
+      <Button
+        class="w-110px"
+        :disabled="!isValidSpecFile"
+        @click="createSpec"
+      >
+        {{ done ? t('createSpec.done') : t('createSpec.createSpec') }}
+      </Button>
+
+      <Button
+        variant="outline"
+        @click="emits('restart')"
+      >
+        {{ t('components.button.cancel') }}
+      </Button>
+    </StandardModalFooter>
+  </span>
+</template>
+
+<script setup lang="ts">
+import { ref, watch, computed } from 'vue'
+import { useI18n } from '@packages/frontend-shared/src/locales/i18n'
+import Input from '@packages/frontend-shared/src/components/Input.vue'
+import Button from '@packages/frontend-shared/src/components/Button.vue'
+// import Error from '@packages/frontend-shared/src/components/Error.vue'
+// import Warning from '@packages/frontend-shared/src/components/Warning.vue'
+import { useVModels } from '@vueuse/core'
+import { gql, useMutation } from '@urql/vue'
+import SpecPatterns from '../../SpecPatterns.vue'
+import { EmptyGeneratorCardStepOneFragment, EmptyGeneratorCardStepOne_MatchSpecFileDocument, EmptyGeneratorCardStepOne_WriteFileDocument } from '../../../generated/graphql'
+import StandardModalFooter from '@packages/frontend-shared/src/components/StandardModalFooter.vue'
+
+const props = defineProps<{
+  title: string,
+  gql: EmptyGeneratorCardStepOneFragment
+}>()
+
+const { t } = useI18n()
+
+gql`
+fragment EmptyGeneratorCardStepOne on Query {
+  currentProject {
+    id
+    config
+  }
+  ...SpecPatterns
+}
+`
+
+gql`
+mutation EmptyGeneratorCardStepOne_MatchSpecFile($specFile: String!) {
+  matchesSpecPattern (specFile: $specFile) 
+}
+`
+
+gql`
+mutation EmptyGeneratorCardStepOne_WriteFile ($file: String!) {
+  writeFileRelativeToProjectRoot(file: $file)
```

Comment:
This mutation has already been accounted for via the `generateSpecFromSource`, and it will return a file which would allow you to proceed to the `GeneratorSuccess` component. All of the other generate spec flows (for a single file) lead to this result, so I don't imagine this one would be any different.

Calling `generateSpecFromSource` with `{ codeGenCandidate: specFile.value, type: 'integration' }` will generate a bare integration test, you can see what it generates [here](https://github.com/cypress-io/cypress/blob/10.0-release/packages/data-context/src/codegen/templates/integration/integration.ejs). That was the initial requirements for an empty spec, if it has changed we can update that file (and I can help make those changes if you want, naming needs to be updated since integration -> e2e).